src/util/file_util.cpp

```diff
@@ -77,6 +78,43 @@ std::string get_current_working_directory()
   return working_directory;
 }
 
+/// \param rel_path: relative path
+/// \return absolute path
+std::string get_absolute_path(const std::string &rel_path)
+{
+#ifndef _WIN32
+  errno = 0;
+  char rp[4096];
```

Comment:
Why 4096?  If it is platform specific then use the macro for it.

Reply:
Changed the code to remove this.